pvlib/tests/test_snow.py

```diff
@@ -19,6 +19,18 @@ def test_fully_covered_nrel():
     assert_series_equal(expected, fully_covered)
 
 
+def test_fully_covered_nrel_with_snow_depth():
+    dt = pd.date_range(start="2019-1-1 12:00:00", end="2019-1-1 18:00:00",
+                       freq='1h')
+    snowfall_data = pd.Series([1, 5, .6, 4, .23, -5, 19], index=dt)
+    snow_depth = pd.Series([0., 1, 6, 6.6, 10.6, 10., -2], index=dt)
```

Comment:
I guess if a negative snow_depth is given, it is handled as 0, right? Maybe there should be a warning about this. Or should an error be raised instead?

Reply:
I think in general pvlib doesn't do quality checks of user input data unless bad data affects the output of the function.